react/index.d.ts

```diff
@@ -270,9 +276,9 @@ declare namespace React {
     //
     // Event System
     // ----------------------------------------------------------------------
-    interface SyntheticEventBase<CURRENT, TARGET> {
+    interface SyntheticEventBase<Current, Target> {
```

Comment:
I think this is still a reasonable thing to do, if you take my suggestion below of using intersection type to add `{ target: T }`.